src/fhirtypes/ElementDefinition.ts

```diff
@@ -269,6 +271,40 @@ export class ElementDefinition {
       throw new WideningCardinalityError(this.min, this.max, min, max);
     }
 
+    // Sliced elements and slices have special card rules described here:
+    // http://www.hl7.org/fhiR/profiling.html#slice-cardinality
+    // If element is slice definition
+    if (this.slicing) {
+      const slices = this.structDef.elements.filter(e => e.id !== this.id && e.path === this.path);
+      const sumOfMins = slices.reduce((prev, curr) => (prev += curr.min), 0);
+      // Check that new max >= sum of mins of children
+      if (!isUnbounded && sumOfMins > maxInt) {
+        throw new InvalidSumOfSliceMinsError(sumOfMins, max, this.id);
+      }
+      // Check that new max >= every individual child max
+      const overMaxChild = slices.find(child => child.max === '*' || parseInt(child.max) > maxInt);
+      if (!isUnbounded && overMaxChild) {
+        throw new InvalidMaxOfSliceError(overMaxChild.max, overMaxChild.sliceName, max);
+      }
+    }
+
+    // If element is a slice
+    const slicedElement = this.structDef.elements.find(e => e.path === this.path && e.slicing);
+    if (slicedElement) {
+      const slices = this.structDef.elements.filter(
+        e => e.id !== this.id && e.path === this.path && !e.slicing
+      );
+      const sumOfMins = min + slices.reduce((prev, curr) => (prev += curr.min), 0);
+      // Check that slicedElement max >= new sum of mins
+      if (slicedElement.max !== '*' && sumOfMins > parseInt(slicedElement.max)) {
+        throw new InvalidSumOfSliceMinsError(sumOfMins, slicedElement.max, slicedElement.id);
+      }
```

Comment:
I think the above 8 lines of code is extremely similar to lines 278 - 283 above.  Can these be consolidated into a single function?